sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py

```diff
@@ -184,4 +193,8 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}({var_dict})"
 
     def __str__(self) -> str:
-        return self.__repr__()
+        if self.print_as_yaml:
+            yaml_serialized = self._to_dict()
+            return dump_yaml(yaml_serialized, default_flow_style=False)
+        else:
+            return self.__repr__()
```

Comment:
It's prob never a good idea to show __repr__ for __str__. E,g. class name shouldn't be part of __str__.

Once we implement DictMixin for all entity classes. I'm wondering if we should just print the dictionary for __str__.

Reply:
That's true, but I can't remove the use of repr easily for two reasons:
 - It was the existing implementation we already had.
 - A bunch of tests have been built to rely on the output it produces.